third_party/move/tools/move-mutator/src/output.rs

```diff
@@ -79,10 +71,21 @@ pub(crate) fn setup_mutant_path(
         .ok_or(anyhow::anyhow!("Cannot get file stem of {file_path:?}"))?;
 
     // Deal with the file as OsString to avoid problems with non-UTF8 characters.
-    let mut filename = filename.to_os_string();
-    filename.push(OsString::from(format!("_{index}.move")));
+    let filename = filename.to_os_string();
+    for i in 0u32.. {
+        let mut mutant_path = filename.clone();
+        mutant_path.push(OsString::from(format!("_{i}.move")));
+
+        let mutant_path = output_struct.join(mutant_path);
+        if !mutant_path.exists() {
+            return Ok(mutant_path);
+        }
+    }
 
-    Ok(output_struct.join(filename))
+    Err(anyhow::anyhow!(
+        "There is more than {} mutants in {output_dir:?}",
+        u32::MAX.to_string()
```

Comment:
not sure - but this doesn't work?
```suggestion
        u32::MAX
```

Reply:
Done.